test-suite/Makefile.am

```diff
@@ -133,10 +134,10 @@ VirtualDeleteOperator_SOURCES = \
 	VirtualDeleteOperator.cc \
 	stub_libmem.cc
 
-squid-conf-tests: $(top_builddir)/src/squid.conf.default $(srcdir)/squidconf/*
+squid-conf-tests: $(top_builddir)/src/squid.conf.default $(srcdir)/squidconf/*.conf
```

Comment:
This dependency change is wrong: The test should also be redone if any of the `.instructions` files change. I want to  restore/fix this dependency to include all files (and also include the new test-squid-conf.sh script), but that will require more changes like skipping non-config files in the loop below.

However, before I do all that and make the diff look more complicated, I wanted to get the general feedback about this PR overall direction _and_ to ask whether I should move the `echo FAIL/PASS` functionality below to the new test-squid-conf.sh script as well -- it is much easier to write that code in shell than in shell-inside-Makefile-rule, of course, but it will increase the diff further.

Reply:
> @rousskov If there are no more big changes from you I think this can be made a non-Draft and marked for Anubis to merge.

Thank you but the dependency problem described in this change request should be fixed first. There is no reason to introduce this bug in this PR. Since you like the overall PR approach, I can now focus on this fix. I will remove Draft designation once I push that fix.

Reply:
> the dependency problem described in this change request should be fixed first.

Done in 8bf0505. Will merge in a few days unless there are any last-minute objections. Please feel free to clear for merge sooner.